projects/plugins/jetpack/extensions/shared/memberships/settings.js

```diff
@@ -287,23 +288,38 @@ export function NewsletterAccessDocumentSettings( { accessLevel } ) {
 	);
 }
 
-export function NewsletterAccessPrePublishSettings( { accessLevel } ) {
-	const postVisibility = useSelect( select => select( editorStore ).getEditedPostVisibility() );
+export function NewsletterEmailDocumentSettings() {
+	const isPostPublished = useSelect( select => select( editorStore ).isCurrentPostPublished(), [] );
+	const postType = useSelect( select => select( editorStore ).getCurrentPostType(), [] );
+	const [ postMeta, setPostMeta ] = useEntityProp( 'postType', postType, 'meta' );
 
-	const showMisconfigurationWarning = getShowMisconfigurationWarning( postVisibility, accessLevel );
-	const _accessLevel = accessLevel ?? accessOptions.everybody.key;
+	const toggleSendEmail = value => {
+		const postMetaUpdate = {
+			...postMeta,
+			// Meta value is negated, "don't send", but toggle is truthy when enabled "send"
+			[ META_NAME_FOR_POST_DONT_EMAIL_TO_SUBS ]: ! value,
+		};
+		setPostMeta( postMetaUpdate );
+	};
+
+	const isSendEmailEnabled = useSelect( select => {
+		const meta = select( editorStore ).getEditedPostAttribute( 'meta' );
+		// Meta value is negated, "don't send", but toggle is truthy when enabled "send"
+		return ! meta?.[ META_NAME_FOR_POST_DONT_EMAIL_TO_SUBS ];
+	} );
 
 	return (
 		<PostVisibilityCheck
-			render={ () => (
-				<PanelRow className="edit-post-post-visibility">
-					{ showMisconfigurationWarning ? (
-						<MisconfigurationWarning />
-					) : (
-						<SubscribersAffirmation prePublish={ true } accessLevel={ _accessLevel } />
-					) }
-				</PanelRow>
-			) }
+			render={ ( { canEdit } ) => {
+				return (
+					<ToggleControl
+						checked={ isSendEmailEnabled }
+						disabled={ isPostPublished || ! canEdit }
```

Comment:
In a follow-up we should check if the post was published already previously. Prior art:

https://github.com/Automattic/jetpack/blob/2f325aecc2ea5a7d9ec599c8ec9d2c7bf7417909/projects/plugins/jetpack/modules/subscriptions.php#L1047-L1062

https://github.com/Automattic/jetpack/blob/2f325aecc2ea5a7d9ec599c8ec9d2c7bf7417909/projects/plugins/jetpack/extensions/blocks/subscriptions/panel.js#L110-L114

I tried it briefly but it felt flaky, so better in separate PR.

Reply:
https://github.com/Automattic/jetpack/pull/34711